syntax = "proto3";
option optimize_for = SPEED;
package cn.csuosa.pojo.response;
option java_package = "cn.csuosa.chatroom.proto";
option java_outer_classname = "Response";

message ResponsePOJO
{
  enum Type {
    RESULT = 0;
    MESSAGE = 1;
    ChannelList = 2;
    ChannelMemberList = 3;
  }

  Type type = 1;
  repeated Message message = 2;
  optional Result result = 3;
  repeated ChannelInfo channelInfo = 4;
  repeated string memberNick = 5;
}

message Result {
  bool result = 1;
  string msg = 2;
}

message Message {
  string recTime = 1;
  string channel = 2;
  string fromNick = 3;
  string content = 4;
}

message ChannelInfo {
  string name = 1;
  bool isPublic = 2;
  int32 memberNum = 3;
}